code/script.js

```diff
@@ -1,51 +1,312 @@
+      setTimeout(() => goodbye('room-service'), 1800)
+    }) 
+
+    document.getElementById('sallad').addEventListener('click', () => {
+      showMessage(`I would love a Ceasar Salad`, 'user')
+      setTimeout(() => goodbye('room-service'), 1800)
+    })
+
+    document.getElementById('tomatoesoup').addEventListener('click', () => {
+      showMessage(`I crave some Tomatoe Soup`, 'user')
+      setTimeout(() => goodbye('room-service'), 1800)
+    })
+  }
+
+  else if (selection === 'wake-up') {        // If guest needs a wake-up call
+    showMessage('Sure, at what time?', 'bot');
+    inputWrapper.innerHTML = `
+      <form id="wakeUpForm"> 
+        <input id="wakeUpInput" type="text" placeholder="wake-up time.."/>  
+        <button id="send-btn" class="send-btn" type="submit"> 
+           Send
+        </button>
+      </form>
+    `
+  }
+  
+      document.getElementById('wakeUpForm').addEventListener('submit', (event) => {
+      event.preventDefault();
+      const wakeUp = wakeUpInput.value
+      showMessage(`Wake me up at ${wakeUp}`, 'user');
+      wakeUpInput.value = '';
+      
+      if (wakeUp === '') {       // this will prompt the user to answer correctly
+      setTimeout(() => showMessage(`Please, give me a time.`, 'bot'), 1500);
+      } else {
+      setTimeout(()=> goodbye('wake-up'), 1800);
+  }
+})
+
 }
 
+// Question 4 - Bot sends thank you message and ask 'Do you need anything else?'
+
+const goodbye = (selection) => {
+
+  if (selection === 'amenities') {
+    showMessage(`We will send up those amenities for you right away. Do you need anything else?`, 'bot')
+    inputWrapper.innerHTML = `
+    <button id="yes" type="sbumit" class="chat-btn">Yes, please </button>
+    <button id="no" type="submit" class="chat-btn">No, thank you</button>
+  `
+   
+  document.getElementById('yes').addEventListener('click', () => {      // If guest says yes
+    showMessage(`Yes, I need something else`, 'user')
+    setTimeout(() => helpSelection(), 1500)
+  })
+
+  document.getElementById('no').addEventListener('click', () => {     // If guest says no 
+    showMessage(`No, thank you`, 'user')
+    setTimeout(() => confirmation(), 1500)
+
+  })
+
+   }
+
+  else if (selection === 'room-service') {
+    showMessage(`Great choice! We will serve you room-service shortly. Do you need anything else?`, 'bot')
+    inputWrapper.innerHTML = `
+    <button id="yes" type="sbumit" class="chat-btn">Yes, please </button>
+    <button id="no" type="submit" class="chat-btn">No, thank you</button>
+  `
+
+  document.getElementById('yes').addEventListener('click', () => {   // If guest says yes
+    showMessage(`Yes, I need something else`, 'user')
+    setTimeout(() => helpSelection(), 1500)
+  }) 
+
+  document.getElementById('no').addEventListener('click', () => {    // If guest says no 
+    showMessage(`No, thank you`, 'user')
+    setTimeout(() => confirmation(), 1500)
+
+  })
+
+    
+  }
+
+  else if (selection === 'wake-up') {
```

Comment:
Why not use "else" instead of "else if"?

Reply:
using else if is because we're giving it a condition, else returns a function that applies to everything that doesn't have the condition assigned to if